singleuser/Dockerfile

```diff
@@ -0,0 +1,24 @@
+ARG JUPYTERHUB_VERSION
+FROM quay.io/jupyter/minimal-notebook:hub-${JUPYTERHUB_VERSION}
+
+COPY ./config /tmp/config
+
+USER root
+RUN \
+    # Install dependencies: apt
+    bash /tmp/config/apt_install.sh \
+    # Add startup hook
+    && cp /tmp/config/before_start_hook.sh /usr/local/bin/before-notebook.d/ \
+    && chmod +x /usr/local/bin/before-notebook.d/before_start_hook.sh \
+    # Add jupyter*config*.py
+    && cp /tmp/config/jupyter*config*.py /etc/jupyter/
+
+ARG PYTHON_VERSION
+RUN \
+    # remove default work directory
+    [ -d "/home/jovyan/work" ] && rm -r /home/jovyan/work \
+    # Install dependencies: pip and conda
+    && conda install -y -n base python=${PYTHON_VERSION} \
+    && pip install -r /tmp/config/pip_requirements.txt \
+    && conda install --yes --file /tmp/config/conda_requirements.txt \
```

Comment:
When I remove the `conda install`, build succeeds but we get the error:

```console
❯ dkc -f singleuser/docker-compose.yaml up --build singleuser    
[+] Building 0.4s (9/9) FINISHED                                                                                            docker:default
 => [singleuser internal] load build definition from Dockerfile                                                                       0.0s
 => => transferring dockerfile: 882B                                                                                                  0.0s
 => [singleuser internal] load metadata for quay.io/jupyter/minimal-notebook:hub-4.0.2                                                0.2s
 => [singleuser internal] load .dockerignore                                                                                          0.0s
 => => transferring context: 2B                                                                                                       0.0s
 => [singleuser internal] load build context                                                                                          0.0s
 => => transferring context: 388B                                                                                                     0.0s
 => [singleuser 1/4] FROM quay.io/jupyter/minimal-notebook:hub-4.0.2@sha256:628aee34f9cb673f9a90489788d0052a9f2e311ed72fc23b8f5918db  0.0s
 => CACHED [singleuser 2/4] COPY ./config /tmp/config                                                                                 0.0s
 => CACHED [singleuser 3/4] RUN     bash /tmp/config/apt_install.sh     && cp /tmp/config/before_start_hook.sh /usr/local/bin/before  0.0s
 => CACHED [singleuser 4/4] RUN     [ -d "/home/jovyan/work" ] && rm -r /home/jovyan/work     && pip install -r /tmp/config/pip_requ  0.0s
 => [singleuser] exporting to image                                                                                                   0.0s
 => => exporting layers                                                                                                               0.0s
 => => writing image sha256:ae5090756d458ccaeaa1d7f4a2b42c661aa91b035bdeafefea94f52f280914e4                                          0.0s
 => => naming to docker.io/datajoint/djlabhub:singleuser-4.0.2-py3.10                                                                 0.0s
[+] Running 1/0
 ✔ Container djlabhub-singleuser  Created                                                                                             0.0s 
Attaching to djlabhub-singleuser
djlabhub-singleuser  | Entered start.sh with args: start-notebook.py
djlabhub-singleuser  | Running hooks in: /usr/local/bin/start-notebook.d as uid: 0 gid: 0
djlabhub-singleuser  | Done running hooks in: /usr/local/bin/start-notebook.d
djlabhub-singleuser  | Running hooks in: /usr/local/bin/before-notebook.d as uid: 0 gid: 0
djlabhub-singleuser  | Sourcing shell script: /usr/local/bin/before-notebook.d/10activate-conda-env.sh
djlabhub-singleuser  | Sourcing shell script: /usr/local/bin/before-notebook.d/before_start_hook.sh
djlabhub-singleuser  | INFO::Datajoint Startup Hook
djlabhub-singleuser  | INFO::Changing Markdown Preview to preview as default
djlabhub-singleuser  | INFO::Cloning repo https://github.com/datajoint/datajoint-tutorials.git
djlabhub-singleuser  | fatal: destination path '/home/jovyan/datajoint-tutorials' already exists and is not an empty directory.
djlabhub-singleuser exited with code 128
```

Reply:
I added there because previously we had some wired dependencies that can only be installed by conda, but I don't think we need it anymore. I'll take the conda install out

Reply:
@ethho Updated, please take a look

Reply:
Still failing, the conda lines are still in the Dockerfile:

```console
❯ JUPYTERHUB_VERSION=4.0.2 PYTHON_VERSION=3.10 docker compose -f singleuser/docker-compose.yaml up --build singleuser
[+] Building 28.6s (8/8) FINISHED                                                                                           docker:default
 => [singleuser internal] load build definition from Dockerfile                                                                       0.0s
 => => transferring dockerfile: 881B                                                                                                  0.0s
 => [singleuser internal] load metadata for quay.io/jupyter/minimal-notebook:hub-4.0.2                                                0.7s
 => [singleuser internal] load .dockerignore                                                                                          0.0s
 => => transferring context: 2B                                                                                                       0.0s
 => [singleuser 1/4] FROM quay.io/jupyter/minimal-notebook:hub-4.0.2@sha256:c31664d79087a351962a8d411576a52b0ded36869d3ca2dc7be75e65  0.0s
 => [singleuser internal] load build context                                                                                          0.0s
 => => transferring context: 388B                                                                                                     0.0s
 => CACHED [singleuser 2/4] COPY ./config /tmp/config                                                                                 0.0s
 => CACHED [singleuser 3/4] RUN     bash /tmp/config/apt_install.sh     && cp /tmp/config/before_start_hook.sh /usr/local/bin/before  0.0s
 => ERROR [singleuser 4/4] RUN     [ -d "/home/jovyan/work" ] && rm -r /home/jovyan/work     && conda install -y -n base python=3.1  27.7s
------                                                                                                                                     
 > [singleuser 4/4] RUN     [ -d "/home/jovyan/work" ] && rm -r /home/jovyan/work     && conda install -y -n base python=3.10     && pip install -r /tmp/config/pip_requirements.txt     && conda install --yes --file /tmp/config/conda_requirements.txt     && conda clean --all -f -y:                                                                                                                                        
1.918 Channels:                                                                                                                            
1.918  - conda-forge                                                                                                                       
1.918 Platform: linux-64
1.918 Collecting package metadata (repodata.json): ...working... done
26.86 Solving environment: ...working... failed
26.94 
26.94 SpecsConfigurationConflictError: Requested specs conflict with configured specs.
26.94   requested specs: 
26.94     - python=3.10
26.94   pinned specs: 
26.94     - python=3.11
26.94 Use 'conda config --show-sources' to look for 'pinned_specs' and 'track_features'
26.94 configuration parameters.  Pinned specs may also be defined in the file
26.94 /opt/conda/conda-meta/pinned.
26.94 
26.94 
------
failed to solve: process "/bin/bash -o pipefail -c [ -d \"/home/jovyan/work\" ] && rm -r /home/jovyan/work     && conda install -y -n base python=${PYTHON_VERSION}     && pip install -r /tmp/config/pip_requirements.txt     && conda install --yes --file /tmp/config/conda_requirements.txt     && conda clean --all -f -y" did not complete successfully: exit code: 1
```

Reply:
@ethho Sorry, I forgot to push my changes...

Reply:
@yambottle This build is still failing on my machine (x86, though in theory arch shouldn't matter with Docker?). Since you can presumably build and push the image on your machine, I'll approve this PR to unblock. Installing `scikit-image` using conda or apt instead of pip might work better, since these distributions typically ship the built library instead of compiling as part of setup.py

<details><summary>Error Log</summary>
<p>

```console
# ...
djlabhub-singleuser  |       INFO: compile options: '-I/opt/conda/include/python3.11 -c'
djlabhub-singleuser  |       extra options: '-msse -msse2 -Werror'
djlabhub-singleuser  |       WARN: CCompilerOpt.dist_test[637] : CCompilerOpt._dist_test_spawn[771] : Command (gcc -pthread -B /opt/conda/compiler_compat -DNDEBUG -fwrapv -O2 -Wall -fPIC -O2 -isystem /opt/conda/include -fPIC -O2 -isystem /opt/conda/include -fPIC -I/opt/conda/include/python3.11 -c /tmp/pip-build-env-g3vfvlx7/overlay/lib/python3.11/site-packages/numpy/distutils/checks/cpu_sse.c -o /tmp/tmpzpuol27x/tmp/pip-build-env-g3vfvlx7/overlay/lib/python3.11/site-packages/numpy/distutils/checks/cpu_sse.o -MMD -MF /tmp/tmpzpuol27x/tmp/pip-build-env-g3vfvlx7/overlay/lib/python3.11/site-packages/numpy/distutils/checks/cpu_sse.o.d -msse -msse2 -Werror) failed with exit status 1 output ->
djlabhub-singleuser  |       In file included from /usr/lib/gcc/x86_64-linux-gnu/11/include/xmmintrin.h:34,
djlabhub-singleuser  |                        from /tmp/pip-build-env-g3vfvlx7/overlay/lib/python3.11/site-packages/numpy/distutils/checks/cpu_sse.c:14:
djlabhub-singleuser  |       /usr/lib/gcc/x86_64-linux-gnu/11/include/mm_malloc.h:27:10: fatal error: stdlib.h: No such file or directory
djlabhub-singleuser  |          27 | #include <stdlib.h>
djlabhub-singleuser  |             |          ^~~~~~~~~~
djlabhub-singleuser  |       compilation terminated.
djlabhub-singleuser  |       
djlabhub-singleuser  |       WARN: CCompilerOpt.feature_test[1576] : testing failed
djlabhub-singleuser  |       INFO: CCompilerOpt.feature_test[1560] : testing feature 'SSE2' with flags (-msse -msse2)
djlabhub-singleuser  |       INFO: C compiler: gcc -pthread -B /opt/conda/compiler_compat -DNDEBUG -fwrapv -O2 -Wall -fPIC -O2 -isystem /opt/conda/include -fPIC -O2 -isystem /opt/conda/include -fPIC
djlabhub-singleuser  |       
djlabhub-singleuser  |       INFO: compile options: '-I/opt/conda/include/python3.11 -c'
djlabhub-singleuser  |       extra options: '-msse -msse2 -Werror'
djlabhub-singleuser  |       WARN: CCompilerOpt.dist_test[637] : CCompilerOpt._dist_test_spawn[771] : Command (gcc -pthread -B /opt/conda/compiler_compat -DNDEBUG -fwrapv -O2 -Wall -fPIC -O2 -isystem /opt/conda/include -fPIC -O2 -isystem /opt/conda/include -fPIC -I/opt/conda/include/python3.11 -c /tmp/pip-build-env-g3vfvlx7/overlay/lib/python3.11/site-packages/numpy/distutils/checks/cpu_sse2.c -o /tmp/tmpzpuol27x/tmp/pip-build-env-g3vfvlx7/overlay/lib/python3.11/site-packages/numpy/distutils/checks/cpu_sse2.o -MMD -MF /tmp/tmpzpuol27x/tmp/pip-build-env-g3vfvlx7/overlay/lib/python3.11/site-packages/numpy/distutils/checks/cpu_sse2.o.d -msse -msse2 -Werror) failed with exit status 1 output ->
djlabhub-singleuser  |       In file included from /usr/lib/gcc/x86_64-linux-gnu/11/include/xmmintrin.h:34,
djlabhub-singleuser  |                        from /usr/lib/gcc/x86_64-linux-gnu/11/include/emmintrin.h:31,
djlabhub-singleuser  |                        from /tmp/pip-build-env-g3vfvlx7/overlay/lib/python3.11/site-packages/numpy/distutils/checks/cpu_sse2.c:14:
djlabhub-singleuser  |       /usr/lib/gcc/x86_64-linux-gnu/11/include/mm_malloc.h:27:10: fatal error: stdlib.h: No such file or directory
djlabhub-singleuser  |          27 | #include <stdlib.h>
djlabhub-singleuser  |             |          ^~~~~~~~~~
djlabhub-singleuser  |       compilation terminated.
djlabhub-singleuser  |       
djlabhub-singleuser  |       WARN: CCompilerOpt.feature_test[1576] : testing failed
djlabhub-singleuser  |       INFO: CCompilerOpt.__init__[1804] : check requested dispatch-able features
djlabhub-singleuser  |       INFO: CCompilerOpt.cc_test_flags[1086] : testing flags (-msse3)
djlabhub-singleuser  |       INFO: C compiler: gcc -pthread -B /opt/conda/compiler_compat -DNDEBUG -fwrapv -O2 -Wall -fPIC -O2 -isystem /opt/conda/include -fPIC -O2 -isystem /opt/conda/include -fPIC
djlabhub-singleuser  |       
djlabhub-singleuser  |       INFO: compile options: '-I/opt/conda/include/python3.11 -c'
djlabhub-singleuser  |       extra options: '-msse3'
djlabhub-singleuser  |       INFO: CCompilerOpt.cc_test_flags[1086] : testing flags (-mssse3)
djlabhub-singleuser  |       INFO: C compiler: gcc -pthread -B /opt/conda/compiler_compat -DNDEBUG -fwrapv -O2 -Wall -fPIC -O2 -isystem /opt/conda/include -fPIC -O2 -isystem /opt/conda/include -fPIC
djlabhub-singleuser  |       
djlabhub-singleuser  |       INFO: compile options: '-I/opt/conda/include/python3.11 -c'
djlabhub-singleuser  |       extra options: '-mssse3'
djlabhub-singleuser  |       INFO: CCompilerOpt.cc_test_flags[1086] : testing flags (-msse4.1)
djlabhub-singleuser  |       INFO: C compiler: gcc -pthread -B /opt/conda/compiler_compat -DNDEBUG -fwrapv -O2 -Wall -fPIC -O2 -isystem /opt/conda/include -fPIC -O2 -isystem /opt/conda/include -fPIC
djlabhub-singleuser  |       
djlabhub-singleuser  |       INFO: compile options: '-I/opt/conda/include/python3.11 -c'
djlabhub-singleuser  |       extra options: '-msse4.1'
djlabhub-singleuser  |       INFO: CCompilerOpt.feature_test[1560] : testing feature 'SSE41' with flags (-msse -msse2 -msse3 -mssse3 -msse4.1)
djlabhub-singleuser  |       INFO: C compiler: gcc -pthread -B /opt/conda/compiler_compat -DNDEBUG -fwrapv -O2 -Wall -fPIC -O2 -isystem /opt/conda/include -fPIC -O2 -isystem /opt/conda/include -fPIC
djlabhub-singleuser  |       
djlabhub-singleuser  |       INFO: compile options: '-I/opt/conda/include/python3.11 -c'
djlabhub-singleuser  |       extra options: '-msse -msse2 -msse3 -mssse3 -msse4.1 -Werror'
djlabhub-singleuser  |       WARN: CCompilerOpt.dist_test[637] : CCompilerOpt._dist_test_spawn[771] : Command (gcc -pthread -B /opt/conda/compiler_compat -DNDEBUG -fwrapv -O2 -Wall -fPIC -O2 -isystem /opt/conda/include -fPIC -O2 -isystem /opt/conda/include -fPIC -I/opt/conda/include/python3.11 -c /tmp/pip-build-env-g3vfvlx7/overlay/lib/python3.11/site-packages/numpy/distutils/checks/cpu_sse41.c -o /tmp/tmpzpuol27x/tmp/pip-build-env-g3vfvlx7/overlay/lib/python3.11/site-packages/numpy/distutils/checks/cpu_sse41.o -MMD -MF /tmp/tmpzpuol27x/tmp/pip-build-env-g3vfvlx7/overlay/lib/python3.11/site-packages/numpy/distutils/checks/cpu_sse41.o.d -msse -msse2 -msse3 -mssse3 -msse4.1 -Werror) failed with exit status 1 output ->
djlabhub-singleuser  |       In file included from /usr/lib/gcc/x86_64-linux-gnu/11/include/xmmintrin.h:34,
djlabhub-singleuser  |                        from /usr/lib/gcc/x86_64-linux-gnu/11/include/emmintrin.h:31,
djlabhub-singleuser  |                        from /usr/lib/gcc/x86_64-linux-gnu/11/include/pmmintrin.h:31,
djlabhub-singleuser  |                        from /usr/lib/gcc/x86_64-linux-gnu/11/include/tmmintrin.h:31,
djlabhub-singleuser  |                        from /usr/lib/gcc/x86_64-linux-gnu/11/include/smmintrin.h:32,
djlabhub-singleuser  |                        from /tmp/pip-build-env-g3vfvlx7/overlay/lib/python3.11/site-packages/numpy/distutils/checks/cpu_sse41.c:14:
djlabhub-singleuser  |       /usr/lib/gcc/x86_64-linux-gnu/11/include/mm_malloc.h:27:10: fatal error: stdlib.h: No such file or directory
djlabhub-singleuser  |          27 | #include <stdlib.h>
djlabhub-singleuser  |             |          ^~~~~~~~~~
djlabhub-singleuser  |       compilation terminated.
djlabhub-singleuser  |       
djlabhub-singleuser  |       WARN: CCompilerOpt.feature_test[1576] : testing failed
djlabhub-singleuser  |       INFO: CCompilerOpt.feature_test[1560] : testing feature 'SSE3' with flags (-msse -msse2 -msse3)
djlabhub-singleuser  |       INFO: C compiler: gcc -pthread -B /opt/conda/compiler_compat -DNDEBUG -fwrapv -O2 -Wall -fPIC -O2 -isystem /opt/conda/include -fPIC -O2 -isystem /opt/conda/include -fPIC
djlabhub-singleuser  |       
djlabhub-singleuser  |       INFO: compile options: '-I/opt/conda/include/python3.11 -c'
djlabhub-singleuser  |       extra options: '-msse -msse2 -msse3 -Werror'
djlabhub-singleuser  |       WARN: CCompilerOpt.dist_test[637] : CCompilerOpt._dist_test_spawn[771] : Command (gcc -pthread -B /opt/conda/compiler_compat -DNDEBUG -fwrapv -O2 -Wall -fPIC -O2 -isystem /opt/conda/include -fPIC -O2 -isystem /opt/conda/include -fPIC -I/opt/conda/include/python3.11 -c /tmp/pip-build-env-g3vfvlx7/overlay/lib/python3.11/site-packages/numpy/distutils/checks/cpu_sse3.c -o /tmp/tmpzpuol27x/tmp/pip-build-env-g3vfvlx7/overlay/lib/python3.11/site-packages/numpy/distutils/checks/cpu_sse3.o -MMD -MF /tmp/tmpzpuol27x/tmp/pip-build-env-g3vfvlx7/overlay/lib/python3.11/site-packages/numpy/distutils/checks/cpu_sse3.o.d -msse -msse2 -msse3 -Werror) failed with exit status 1 output ->
djlabhub-singleuser  |       In file included from /usr/lib/gcc/x86_64-linux-gnu/11/include/xmmintrin.h:34,
djlabhub-singleuser  |                        from /usr/lib/gcc/x86_64-linux-gnu/11/include/emmintrin.h:31,
djlabhub-singleuser  |                        from /usr/lib/gcc/x86_64-linux-gnu/11/include/pmmintrin.h:31,
djlabhub-singleuser  |                        from /tmp/pip-build-env-g3vfvlx7/overlay/lib/python3.11/site-packages/numpy/distutils/checks/cpu_sse3.c:14:
djlabhub-singleuser  |       /usr/lib/gcc/x86_64-linux-gnu/11/include/mm_malloc.h:27:10: fatal error: stdlib.h: No such file or directory
djlabhub-singleuser  |          27 | #include <stdlib.h>
djlabhub-singleuser  |             |          ^~~~~~~~~~
djlabhub-singleuser  |       compilation terminated.
djlabhub-singleuser  |       
djlabhub-singleuser  |       WARN: CCompilerOpt.feature_test[1576] : testing failed
djlabhub-singleuser  |       INFO: CCompilerOpt.cc_test_flags[1086] : testing flags (-mpopcnt)
djlabhub-singleuser  |       INFO: C compiler: gcc -pthread -B /opt/conda/compiler_compat -DNDEBUG -fwrapv -O2 -Wall -fPIC -O2 -isystem /opt/conda/include -fPIC -O2 -isystem /opt/conda/include -fPIC
djlabhub-singleuser  |       
djlabhub-singleuser  |       INFO: compile options: '-I/opt/conda/include/python3.11 -c'
djlabhub-singleuser  |       extra options: '-mpopcnt'
djlabhub-singleuser  |       INFO: CCompilerOpt.cc_test_flags[1086] : testing flags (-msse4.2)
djlabhub-singleuser  |       INFO: C compiler: gcc -pthread -B /opt/conda/compiler_compat -DNDEBUG -fwrapv -O2 -Wall -fPIC -O2 -isystem /opt/conda/include -fPIC -O2 -isystem /opt/conda/include -fPIC
djlabhub-singleuser  |       
djlabhub-singleuser  |       INFO: compile options: '-I/opt/conda/include/python3.11 -c'
djlabhub-singleuser  |       extra options: '-msse4.2'
djlabhub-singleuser  |       INFO: CCompilerOpt.cc_test_flags[1086] : testing flags (-mavx)
djlabhub-singleuser  |       INFO: C compiler: gcc -pthread -B /opt/conda/compiler_compat -DNDEBUG -fwrapv -O2 -Wall -fPIC -O2 -isystem /opt/conda/include -fPIC -O2 -isystem /opt/conda/include -fPIC
djlabhub-singleuser  |       
djlabhub-singleuser  |       INFO: compile options: '-I/opt/conda/include/python3.11 -c'
djlabhub-singleuser  |       extra options: '-mavx'
djlabhub-singleuser  |       INFO: CCompilerOpt.cc_test_flags[1086] : testing flags (-mf16c)
djlabhub-singleuser  |       INFO: C compiler: gcc -pthread -B /opt/conda/compiler_compat -DNDEBUG -fwrapv -O2 -Wall -fPIC -O2 -isystem /opt/conda/include -fPIC -O2 -isystem /opt/conda/include -fPIC
djlabhub-singleuser  |       
djlabhub-singleuser  |       INFO: compile options: '-I/opt/conda/include/python3.11 -c'
djlabhub-singleuser  |       extra options: '-mf16c'
djlabhub-singleuser  |       INFO: CCompilerOpt.cc_test_flags[1086] : testing flags (-mfma)
djlabhub-singleuser  |       INFO: C compiler: gcc -pthread -B /opt/conda/compiler_compat -DNDEBUG -fwrapv -O2 -Wall -fPIC -O2 -isystem /opt/conda/include -fPIC -O2 -isystem /opt/conda/include -fPIC
djlabhub-singleuser  |       
djlabhub-singleuser  |       INFO: compile options: '-I/opt/conda/include/python3.11 -c'
djlabhub-singleuser  |       extra options: '-mfma'
djlabhub-singleuser  |       INFO: CCompilerOpt.feature_test[1560] : testing feature 'FMA3' with flags (-msse -msse2 -msse3 -mssse3 -msse4.1 -mpopcnt -msse4.2 -mavx -mf16c -mfma)
djlabhub-singleuser  |       INFO: C compiler: gcc -pthread -B /opt/conda/compiler_compat -DNDEBUG -fwrapv -O2 -Wall -fPIC -O2 -isystem /opt/conda/include -fPIC -O2 -isystem /opt/conda/include -fPIC
djlabhub-singleuser  |       
djlabhub-singleuser  |       INFO: compile options: '-I/opt/conda/include/python3.11 -c'
djlabhub-singleuser  |       extra options: '-msse -msse2 -msse3 -mssse3 -msse4.1 -mpopcnt -msse4.2 -mavx -mf16c -mfma -Werror'
djlabhub-singleuser  |       WARN: CCompilerOpt.dist_test[637] : CCompilerOpt._dist_test_spawn[771] : Command (gcc -pthread -B /opt/conda/compiler_compat -DNDEBUG -fwrapv -O2 -Wall -fPIC -O2 -isystem /opt/conda/include -fPIC -O2 -isystem /opt/conda/include -fPIC -I/opt/conda/include/python3.11 -c /tmp/pip-build-env-g3vfvlx7/overlay/lib/python3.11/site-packages/numpy/distutils/checks/cpu_fma3.c -o /tmp/tmpzpuol27x/tmp/pip-build-env-g3vfvlx7/overlay/lib/python3.11/site-packages/numpy/distutils/checks/cpu_fma3.o -MMD -MF /tmp/tmpzpuol27x/tmp/pip-build-env-g3vfvlx7/overlay/lib/python3.11/site-packages/numpy/distutils/checks/cpu_fma3.o.d -msse -msse2 -msse3 -mssse3 -msse4.1 -mpopcnt -msse4.2 -mavx -mf16c -mfma -Werror) failed with exit status 1 output ->
djlabhub-singleuser  |       In file included from /usr/lib/gcc/x86_64-linux-gnu/11/include/xmmintrin.h:34,
djlabhub-singleuser  |                        from /tmp/pip-build-env-g3vfvlx7/overlay/lib/python3.11/site-packages/numpy/distutils/checks/cpu_fma3.c:14:
djlabhub-singleuser  |       /usr/lib/gcc/x86_64-linux-gnu/11/include/mm_malloc.h:27:10: fatal error: stdlib.h: No such file or directory
djlabhub-singleuser  |          27 | #include <stdlib.h>
djlabhub-singleuser  |             |          ^~~~~~~~~~
djlabhub-singleuser  |       compilation terminated.
djlabhub-singleuser  |       
djlabhub-singleuser  |       WARN: CCompilerOpt.feature_test[1576] : testing failed
djlabhub-singleuser  |       INFO: CCompilerOpt.feature_test[1560] : testing feature 'AVX' with flags (-msse -msse2 -msse3 -mssse3 -msse4.1 -mpopcnt -msse4.2 -mavx)
djlabhub-singleuser  |       INFO: C compiler: gcc -pthread -B /opt/conda/compiler_compat -DNDEBUG -fwrapv -O2 -Wall -fPIC -O2 -isystem /opt/conda/include -fPIC -O2 -isystem /opt/conda/include -fPIC
djlabhub-singleuser  |       
djlabhub-singleuser  |       INFO: compile options: '-I/opt/conda/include/python3.11 -c'
djlabhub-singleuser  |       extra options: '-msse -msse2 -msse3 -mssse3 -msse4.1 -mpopcnt -msse4.2 -mavx -Werror'
djlabhub-singleuser  |       WARN: CCompilerOpt.dist_test[637] : CCompilerOpt._dist_test_spawn[771] : Command (gcc -pthread -B /opt/conda/compiler_compat -DNDEBUG -fwrapv -O2 -Wall -fPIC -O2 -isystem /opt/conda/include -fPIC -O2 -isystem /opt/conda/include -fPIC -I/opt/conda/include/python3.11 -c /tmp/pip-build-env-g3vfvlx7/overlay/lib/python3.11/site-packages/numpy/distutils/checks/cpu_avx.c -o /tmp/tmpzpuol27x/tmp/pip-build-env-g3vfvlx7/overlay/lib/python3.11/site-packages/numpy/distutils/checks/cpu_avx.o -MMD -MF /tmp/tmpzpuol27x/tmp/pip-build-env-g3vfvlx7/overlay/lib/python3.11/site-packages/numpy/distutils/checks/cpu_avx.o.d -msse -msse2 -msse3 -mssse3 -msse4.1 -mpopcnt -msse4.2 -mavx -Werror) failed with exit status 1 output ->
djlabhub-singleuser  |       In file included from /usr/lib/gcc/x86_64-linux-gnu/11/include/xmmintrin.h:34,
djlabhub-singleuser  |                        from /usr/lib/gcc/x86_64-linux-gnu/11/include/immintrin.h:31,
djlabhub-singleuser  |                        from /tmp/pip-build-env-g3vfvlx7/overlay/lib/python3.11/site-packages/numpy/distutils/checks/cpu_avx.c:14:
djlabhub-singleuser  |       /usr/lib/gcc/x86_64-linux-gnu/11/include/mm_malloc.h:27:10: fatal error: stdlib.h: No such file or directory
djlabhub-singleuser  |          27 | #include <stdlib.h>
djlabhub-singleuser  |             |          ^~~~~~~~~~
djlabhub-singleuser  |       compilation terminated.
djlabhub-singleuser  |       
djlabhub-singleuser  |       WARN: CCompilerOpt.feature_test[1576] : testing failed
djlabhub-singleuser  |       INFO: CCompilerOpt.__init__[1820] : initialize targets groups
djlabhub-singleuser  |       INFO: CCompilerOpt.__init__[1822] : parse target group simd_test
djlabhub-singleuser  |       INFO: CCompilerOpt._parse_target_tokens[2033] : skip targets ((FMA3 AVX2) FMA4 VSX3 SSE2 XOP VX AVX512_SKX SSE42 VSX2 NEON AVX512F VXE2 VXE ASIMD VSX4 VSX) not part of baseline or dispatch-able features
djlabhub-singleuser  |       INFO: CCompilerOpt.generate_dispatch_header[2366] : generate CPU dispatch header: (build/src.linux-x86_64-3.11/numpy/distutils/include/npy_cpu_dispatch_config.h)
djlabhub-singleuser  |       WARN: CCompilerOpt.generate_dispatch_header[2375] : dispatch header dir build/src.linux-x86_64-3.11/numpy/distutils/include does not exist, creating it
djlabhub-singleuser  |       INFO: customize UnixCCompiler
djlabhub-singleuser  |       INFO: customize UnixCCompiler using ConditionalOpenMP
djlabhub-singleuser  |       INFO: C compiler: gcc -pthread -B /opt/conda/compiler_compat -DNDEBUG -fwrapv -O2 -Wall -fPIC -O2 -isystem /opt/conda/include -fPIC -O2 -isystem /opt/conda/include -fPIC
djlabhub-singleuser  |       
djlabhub-singleuser  |       INFO: compile options: '-I/opt/conda/include/python3.11 -c'
djlabhub-singleuser  |       extra options: '-fopenmp'
djlabhub-singleuser  |       INFO: gcc: test.c
djlabhub-singleuser  |       test.c:1:17: warning: extra tokens at end of #include directive
djlabhub-singleuser  |           1 | #include <omp.h>int main(int argc, char** argv) { return(0); }
djlabhub-singleuser  |             |                 ^~~
djlabhub-singleuser  |       INFO: gcc -pthread -B /opt/conda/compiler_compat -shared -Wl,--allow-shlib-undefined -Wl,-rpath,/opt/conda/lib -Wl,-rpath-link,/opt/conda/lib -L/opt/conda/lib -Wl,--allow-shlib-undefined -Wl,-rpath,/opt/conda/lib -Wl,-rpath-link,/opt/conda/lib -L/opt/conda/lib test.o -o libtestlib.so -fopenmp
djlabhub-singleuser  |       /opt/conda/compiler_compat/ld: cannot find crti.o: No such file or directory
djlabhub-singleuser  |       collect2: error: ld returned 1 exit status
djlabhub-singleuser  |       INFO: building 'skimage._shared.geometry' extension
djlabhub-singleuser  |       INFO: compiling C sources
djlabhub-singleuser  |       INFO: C compiler: gcc -pthread -B /opt/conda/compiler_compat -DNDEBUG -fwrapv -O2 -Wall -fPIC -O2 -isystem /opt/conda/include -fPIC -O2 -isystem /opt/conda/include -fPIC
djlabhub-singleuser  |       
djlabhub-singleuser  |       creating build/temp.linux-x86_64-3.11/skimage
djlabhub-singleuser  |       creating build/temp.linux-x86_64-3.11/skimage/_shared
djlabhub-singleuser  |       INFO: compile options: '-I/tmp/pip-build-env-g3vfvlx7/overlay/lib/python3.11/site-packages/numpy/core/include -Ibuild/src.linux-x86_64-3.11/numpy/distutils/include -I/opt/conda/include/python3.11 -c'
djlabhub-singleuser  |       INFO: gcc: skimage/_shared/geometry.c
djlabhub-singleuser  |       In file included from skimage/_shared/geometry.c:19:
djlabhub-singleuser  |       /opt/conda/include/python3.11/Python.h:23:12: fatal error: stdlib.h: No such file or directory
djlabhub-singleuser  |          23 | #  include <stdlib.h>
djlabhub-singleuser  |             |            ^~~~~~~~~~
djlabhub-singleuser  |       compilation terminated.
djlabhub-singleuser  |       error: Command "gcc -pthread -B /opt/conda/compiler_compat -DNDEBUG -fwrapv -O2 -Wall -fPIC -O2 -isystem /opt/conda/include -fPIC -O2 -isystem /opt/conda/include -fPIC -I/tmp/pip-build-env-g3vfvlx7/overlay/lib/python3.11/site-packages/numpy/core/include -Ibuild/src.linux-x86_64-3.11/numpy/distutils/include -I/opt/conda/include/python3.11 -c skimage/_shared/geometry.c -o build/temp.linux-x86_64-3.11/skimage/_shared/geometry.o -MMD -MF build/temp.linux-x86_64-3.11/skimage/_shared/geometry.o.d" failed with exit status 1
djlabhub-singleuser  |       INFO:
djlabhub-singleuser  |       ########### EXT COMPILER OPTIMIZATION ###########
djlabhub-singleuser  |       INFO: Platform      :
djlabhub-singleuser  |         Architecture: x64
djlabhub-singleuser  |         Compiler    : gcc
djlabhub-singleuser  |       
djlabhub-singleuser  |       CPU baseline  :
djlabhub-singleuser  |         Requested   : 'min'
djlabhub-singleuser  |         Enabled     : none
djlabhub-singleuser  |         Flags       : none
djlabhub-singleuser  |         Extra checks: none
djlabhub-singleuser  |       
djlabhub-singleuser  |       CPU dispatch  :
djlabhub-singleuser  |         Requested   : 'max -xop -fma4'
djlabhub-singleuser  |         Enabled     : none
djlabhub-singleuser  |         Generated   : none
djlabhub-singleuser  |       INFO: CCompilerOpt.cache_flush[864] : write cache to path -> /tmp/pip-install-x6ioq4v_/scikit-image_6f43e31248b94ec38fb134e74e35de3e/build/temp.linux-x86_64-3.11/ccompiler_opt_cache_ext.py
djlabhub-singleuser  |       [end of output]
djlabhub-singleuser  |   
djlabhub-singleuser  |   note: This error originates from a subprocess, and is likely not a problem with pip.
djlabhub-singleuser  | Failed to build scikit-image
djlabhub-singleuser  |   ERROR: Failed building wheel for scikit-image
djlabhub-singleuser  | ERROR: Could not build wheels for scikit-image, which is required to install pyproject.toml-based projects
djlabhub-singleuser exited with code 1
```

</p>
</details>